flow/raster_cache.h

```diff
@@ -152,6 +152,14 @@ class RasterCache {
                const SkMatrix& untranslated_matrix,
                const SkPoint& offset = SkPoint());
 
+  // If there is cache entry for this picture, display list or layer, mark it as
+  // used for this frame in order to not get evicted. This is needed during
+  // partial repaint for layers that are outside of current clip and are culled
+  // away.
+  void Touch(SkPicture* picture, const SkMatrix& transformation_matrix);
```

Comment:
Can you please add tests for this behavior?

Have a frame where each of these layers is culled away but raster cache still has them. Can be in `rasterizer_unittests.cc`.